
Checking out the low ortholgous clusters in the mouse data, and how they fit into the conserved subclass annotations.

How distinct are they from the conserved subclasses?
What does the performance look like when using the conserved MetaMarkers to annotate these cells?




```{r}
library(SingleCellExperiment)
library(MetaMarkers)
library(dplyr)
library(ggplot2)
library(ComplexHeatmap)


```


```{r}
celltype_color_palette = c("Chandelier"='#F641A8', 
                           "Pvalb"='#D93137', 
                           "Sst"='#FF9900', 
                           "Sst Chodl"='gold', 
                           "Lamp5"='#DA808C',
                           "Sncg"='#DF70FF',
                           "Vip"='#A45FBF', 
                           "Lamp5 Lhx6"='#935F50',
                           "Pax6"='#71238C',
                           'Serpinf1' = '#A45FBF')


```

First load up the dataset markers, the cross-species metamarkers, and then grab the lists of conserved metamarkers per subclass


```{r}

all_species_markers = list(
    human_10x = read_markers("metamarkers/markers_human_10x.csv.gz"),
    human_ss = read_markers("metamarkers/markers_human_ss.csv.gz"),
    chimp_10x = read_markers("metamarkers/markers_chimp_10x.csv.gz"),
    chimp_ss = read_markers("metamarkers/markers_chimp_ss.csv.gz"),
    gorilla_10x = read_markers("metamarkers/markers_gorilla_10x.csv.gz"),
    gorilla_ss = read_markers("metamarkers/markers_gorilla_ss.csv.gz"),
    macaca_10x = read_markers("metamarkers/markers_macaca_10x.csv.gz"),
    marmoset_10x= read_markers("metamarkers/markers_marmoset_10x.csv.gz"),
    tasic18 = read_markers("metamarkers/ortho_markers_tasic18.csv.gz"),
    zeng_10x_cells = read_markers("metamarkers/ortho_markers_zeng_10x_cells.csv.gz"),
    zeng_10x_nuclei = read_markers("metamarkers/ortho_markers_zeng_10x_nuclei.csv.gz"),
    zeng_smart_nuclei = read_markers("metamarkers/ortho_markers_zeng_smart_nuclei.csv.gz"),
    zeng_smart_cells = read_markers("metamarkers/ortho_markers_zeng_smart_cells.csv.gz"),
    zeng_10xv2_wba = read_markers("metamarkers/ortho_markers_zeng_10xv2_wba.csv.gz"),
    zeng_10xv3_wba = read_markers("metamarkers/ortho_markers_zeng_10xv3_wba.csv.gz"),
    macosko_wba = read_markers("metamarkers/ortho_markers_macosko_wba.csv.gz")
)


all_species_markers$human_10x$cell_type[all_species_markers$human_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_species_markers$human_ss$cell_type[all_species_markers$human_ss$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_species_markers$chimp_10x$cell_type[all_species_markers$chimp_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_species_markers$chimp_ss$cell_type[all_species_markers$chimp_ss$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_species_markers$gorilla_10x$cell_type[all_species_markers$gorilla_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_species_markers$gorilla_ss$cell_type[all_species_markers$gorilla_ss$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_species_markers$macaca_10x$cell_type[all_species_markers$macaca_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_species_markers$marmoset_10x$cell_type[all_species_markers$marmoset_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'


conserved_subclass_meta_markers = make_meta_markers(all_species_markers, detailed_stats = TRUE)

conserved_subclass_meta_markers %>% group_by(cell_type) %>% filter(rank <= 10)


```


```{r}

get_conserved_MetaMarkers = function(meta_markers_df, celltype){
  
  celltype_de = filter(meta_markers_df, cell_type == celltype)
  
  human_check = celltype_de$human_10x | celltype_de$human_ss
  chimp_check = celltype_de$chimp_10x | celltype_de$chimp_ss
  gorilla_check = celltype_de$gorilla_10x | celltype_de$gorilla_ss
  macaca_check = celltype_de$macaca_10x
  marmoset_check = celltype_de$marmoset_10x
  mouse_check = rowSums(celltype_de[ ,c('tasic18','zeng_10x_cells','zeng_10x_nuclei','zeng_smart_cells','zeng_smart_nuclei',
                                     'zeng_10xv2_wba','zeng_10xv3_wba','macosko_wba')]) >= 1
  
  
  cross_species_de_check = human_check & chimp_check & gorilla_check & macaca_check & marmoset_check & mouse_check
  
  conserved_markers = celltype_de$gene[cross_species_de_check]
  
  return(conserved_markers)
}


cons_chandelier_markers = get_conserved_MetaMarkers(conserved_subclass_meta_markers, 'Chandelier')
cons_chandelier_markers = conserved_subclass_meta_markers %>% filter(cell_type == 'Chandelier' & gene %in% cons_chandelier_markers)

cons_pvalb_markers = get_conserved_MetaMarkers(conserved_subclass_meta_markers, 'Pvalb')
cons_pvalb_markers = conserved_subclass_meta_markers %>% filter(cell_type == 'Pvalb' & gene %in% cons_pvalb_markers)

cons_sst_markers = get_conserved_MetaMarkers(conserved_subclass_meta_markers, 'Sst')
cons_sst_markers = conserved_subclass_meta_markers %>% filter(cell_type == 'Sst' & gene %in% cons_sst_markers)

cons_sst_chodl_markers = get_conserved_MetaMarkers(conserved_subclass_meta_markers, 'Sst Chodl')
cons_sst_chodl_markers = conserved_subclass_meta_markers %>% filter(cell_type == 'Sst Chodl' & gene %in% cons_sst_chodl_markers)

cons_vip_markers = get_conserved_MetaMarkers(conserved_subclass_meta_markers, 'Vip')
cons_vip_markers = conserved_subclass_meta_markers %>% filter(cell_type == 'Vip' & gene %in% cons_vip_markers)

cons_sncg_markers = get_conserved_MetaMarkers(conserved_subclass_meta_markers, 'Sncg')
cons_sncg_markers = conserved_subclass_meta_markers %>% filter(cell_type == 'Sncg' & gene %in% cons_sncg_markers)

cons_pax6_markers = get_conserved_MetaMarkers(conserved_subclass_meta_markers, 'Pax6')
cons_pax6_markers = conserved_subclass_meta_markers %>% filter(cell_type == 'Pax6' & gene %in% cons_pax6_markers)

cons_lamp5_markers = get_conserved_MetaMarkers(conserved_subclass_meta_markers, 'Lamp5')
cons_lamp5_markers = conserved_subclass_meta_markers %>% filter(cell_type == 'Lamp5' & gene %in% cons_lamp5_markers)

cons_lamp5_lhx6_markers = get_conserved_MetaMarkers(conserved_subclass_meta_markers, 'Lamp5 Lhx6')
cons_lamp5_lhx6_markers = conserved_subclass_meta_markers %>% filter(cell_type == 'Lamp5 Lhx6' & gene %in% cons_lamp5_lhx6_markers)

all_cons_MetaMarkers = do.call('rbind', list(cons_chandelier_markers, cons_pvalb_markers, cons_sst_markers, cons_sst_chodl_markers, 
                     cons_vip_markers, cons_lamp5_markers, cons_sncg_markers, cons_pax6_markers, cons_lamp5_lhx6_markers))

all_cons_MetaMarkers

```


And this is the mapping between the primate subclasses and the mouse data, with the low/high orthology annotations


```{r}

load(file = '/home/werner/projects/mouse_brain_consensus_taxonomy/data/metaneighbor/avg_mouse_clust_primate_subclass_mappings_gaba.Rdata')
max_avg_mapping

```





And the mouse data with all primate subclass and orthology annotations

```{r}
tasic18 = readRDS('/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/mouse/tasic18_gaba_primate_MN.rds')
zeng_10x_cells = readRDS('/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/mouse/zeng_10x_cells_gaba_primate_MN.rds')
zeng_10x_nuclei = readRDS('/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/mouse/zeng_10x_nuclei_gaba_primate_MN.rds')
zeng_smart_cells = readRDS('/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/mouse/zeng_smart_cells_gaba_primate_MN.rds')
zeng_smart_nuclei = readRDS('/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/mouse/zeng_smart_nuclei_gaba_primate_MN.rds')
zeng_10xv2_wba = readRDS('/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/mouse/zeng_wb_atlas_10xv2_gaba_primate_MN.rds')
zeng_10xv3_wba = readRDS('/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/mouse/zeng_wb_atlas_10xv3_gaba_primate_MN.rds')
macosko_wba = readRDS('/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/mouse/macosko_wb_atlas_gaba_primate_MN.rds')
```


Testing out a quick doublet check with scdblfinder

```{r}

library(scDblFinder)
start = Sys.time()
tasic18 <- scDblFinder(tasic18, clusters = 'cell_cluster')
Sys.time() - start

zeng_10x_cells <- scDblFinder(zeng_10x_cells, clusters = 'cell_cluster')

zeng_10x_nuclei <- scDblFinder(zeng_10x_nuclei, clusters = 'cell_cluster')

zeng_smart_nuclei <- scDblFinder(zeng_smart_nuclei, clusters = 'cell_cluster')

zeng_smart_cells <- scDblFinder(zeng_smart_cells, clusters = 'cell_cluster')

zeng_10xv2_wba <- scDblFinder(zeng_10xv2_wba, clusters = 'cell_cluster')

zeng_10xv3_wba <- scDblFinder(zeng_10xv3_wba, clusters = 'cell_cluster')

macosko_wba <- scDblFinder(macosko_wba, clusters = 'cell_cluster')

table(tasic18$orthology, tasic18$scDblFinder.class)
table(zeng_10x_cells$orthology, zeng_10x_cells$scDblFinder.class)
table(zeng_10x_nuclei$orthology, zeng_10x_nuclei$scDblFinder.class)
table(zeng_smart_nuclei$orthology, zeng_smart_nuclei$scDblFinder.class)
table(zeng_smart_cells$orthology, zeng_smart_cells$scDblFinder.class)
table(zeng_10xv2_wba$orthology, zeng_10xv2_wba$scDblFinder.class)
table(zeng_10xv3_wba$orthology, zeng_10xv3_wba$scDblFinder.class)
table(macosko_wba$orthology, macosko_wba$scDblFinder.class)
```




```{r}


current_metadata = as.data.frame(colData(tasic18))


current_metadata %>% group_by(primate_subclass, orthology, scDblFinder.class) %>% summarise(n = n())

ggplot(current_metadata, aes(x = orthology, y = log10(nCount))) + geom_violin(scale = 'width') +
  facet_wrap(~primate_subclass)

ggplot(current_metadata, aes(x = orthology, y = nFeature)) + geom_violin(scale = 'width') +
  facet_wrap(~primate_subclass)




table(current_metadata$orthology, current_metadata$scDblFinder.class)



```








```{r}
conserved_subclass_meta_markers = make_meta_markers(all_species_markers[names(all_species_markers) != 'tasic18'], detailed_stats = TRUE)

#Predict celltypes with MetaMarkers
#ct_scores = score_cells(log1p(cpm(tasic18)), all_cons_MetaMarkers)

ct_scores = score_cells(log1p(cpm(tasic18)), conserved_subclass_meta_markers %>% filter(rank <= 50))

ct_enrichment = compute_marker_enrichment(ct_scores)
dim(ct_enrichment)

#Max normalizing each cell-type enrichment
ct_enrichment_norm = t(apply(ct_enrichment, 1, function(u) u/max(u)))

ct_pred = assign_cells(ct_scores)

tasic18$MM_predicted_subclass = ct_pred$predicted

tasic18$MM_chandelier_enrich_score = ct_enrichment_norm['all|Chandelier', ]
tasic18$MM_pvalb_enrich_score = ct_enrichment_norm['all|Pvalb', ]
tasic18$MM_sst_enrich_score = ct_enrichment_norm['all|Sst', ]
tasic18$MM_sst_chodl_enrich_score = ct_enrichment_norm['all|Sst Chodl', ]
tasic18$MM_vip_enrich_score = ct_enrichment_norm['all|Vip', ]
tasic18$MM_lamp5_enrich_score = ct_enrichment_norm['all|Lamp5', ]
tasic18$MM_sncg_enrich_score = ct_enrichment_norm['all|Sncg', ]
tasic18$MM_pax6_enrich_score = ct_enrichment_norm['all|Pax6', ]
tasic18$MM_lamp5_lhx6_enrich_score = ct_enrichment_norm['all|Lamp5 Lhx6', ]

scater::plotReducedDim(tasic18, dimred = 'UMAP', color_by = 'orthology') + 
  ggtitle('Orthology annotations') 

scater::plotReducedDim(tasic18, dimred = 'UMAP', color_by = 'primate_subclass') + scale_color_manual(values=celltype_color_palette) + 
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(tasic18, dimred = 'UMAP', color_by = 'MM_predicted_subclass') + scale_color_manual(values=celltype_color_palette) + 
  ggtitle('MM predicted subclass annotations') 

scater::plotReducedDim(tasic18, dimred = 'UMAP', color_by = 'MM_lamp5_enrich_score') +
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(tasic18, dimred = 'UMAP', color_by = 'MM_sncg_enrich_score') +
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(tasic18, dimred = 'UMAP', color_by = 'MM_lamp5_lhx6_enrich_score') +
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(tasic18, dimred = 'UMAP', color_by = 'MM_vip_enrich_score') +
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(tasic18, dimred = 'UMAP', color_by = 'MM_pax6_enrich_score') +
  ggtitle('Primate subclass annotations') 


conserved_subclass_meta_markers = make_meta_markers(all_species_markers[names(all_species_markers) != 'zeng_10x_cells'], detailed_stats = TRUE)

#Predict celltypes with MetaMarkers
#ct_scores = score_cells(log1p(cpm(zeng_10x_cells)), all_cons_MetaMarkers)

ct_scores = score_cells(log1p(cpm(zeng_10x_cells)), conserved_subclass_meta_markers %>% filter(rank <= 50))

ct_enrichment = compute_marker_enrichment(ct_scores)
dim(ct_enrichment)

#Max normalizing each cell-type enrichment
ct_enrichment_norm = t(apply(ct_enrichment, 1, function(u) u/max(u)))

ct_pred = assign_cells(ct_scores)

zeng_10x_cells$MM_predicted_subclass = ct_pred$predicted

zeng_10x_cells$MM_chandelier_enrich_score = ct_enrichment_norm['all|Chandelier', ]
zeng_10x_cells$MM_pvalb_enrich_score = ct_enrichment_norm['all|Pvalb', ]
zeng_10x_cells$MM_sst_enrich_score = ct_enrichment_norm['all|Sst', ]
zeng_10x_cells$MM_sst_chodl_enrich_score = ct_enrichment_norm['all|Sst Chodl', ]
zeng_10x_cells$MM_vip_enrich_score = ct_enrichment_norm['all|Vip', ]
zeng_10x_cells$MM_lamp5_enrich_score = ct_enrichment_norm['all|Lamp5', ]
zeng_10x_cells$MM_sncg_enrich_score = ct_enrichment_norm['all|Sncg', ]
zeng_10x_cells$MM_pax6_enrich_score = ct_enrichment_norm['all|Pax6', ]
zeng_10x_cells$MM_lamp5_lhx6_enrich_score = ct_enrichment_norm['all|Lamp5 Lhx6', ]

scater::plotReducedDim(zeng_10x_cells, dimred = 'UMAP', color_by = 'orthology') + 
  ggtitle('Orthology annotations') 

scater::plotReducedDim(zeng_10x_cells, dimred = 'UMAP', color_by = 'primate_subclass') + scale_color_manual(values=celltype_color_palette) + 
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(zeng_10x_cells, dimred = 'UMAP', color_by = 'MM_predicted_subclass') + scale_color_manual(values=celltype_color_palette) + 
  ggtitle('MM predicted subclass annotations') 

scater::plotReducedDim(zeng_10x_cells, dimred = 'UMAP', color_by = 'MM_lamp5_enrich_score') +
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(zeng_10x_cells, dimred = 'UMAP', color_by = 'MM_sncg_enrich_score') +
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(zeng_10x_cells, dimred = 'UMAP', color_by = 'MM_lamp5_lhx6_enrich_score') +
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(zeng_10x_cells, dimred = 'UMAP', color_by = 'MM_vip_enrich_score') +
  ggtitle('Primate subclass annotations') 

scater::plotReducedDim(zeng_10x_cells, dimred = 'UMAP', color_by = 'MM_pax6_enrich_score') +
  ggtitle('Primate subclass annotations') 


```






For each mouse dataset, make cross-species metamarkers excluding that dataset
Compute the normalized MetaMarker enrichment scores per cluster
  MetaMarkers computes the average expression of each MetaMarker gene set within each cell
  It's enrichment score is the average exp within the cell / average expression across all cells
  Max normalize the enrichment score to compare across cell-type marker sets
  And then to summarize at the cluster level, compute centroids of that Max-normalized enrichment score

And then cluster across all the datasets to compare

```{r}

get_cluster_MM_centroids = function(sce_obj, dataset_name, species = NA, marker_list, num_markers, species_filt = F){
  
  #Cross-species metamarkers excluding the current species
  if(species_filt){
    species_marker_index = !grepl(species, names(marker_list) )
    conserved_subclass_meta_markers = make_meta_markers(marker_list[species_marker_index], detailed_stats = TRUE)
  }else{ #Exclude the specific dataset
    conserved_subclass_meta_markers = make_meta_markers(marker_list[names(marker_list) != dataset_name], detailed_stats = TRUE)
  }

  #Predict celltypes with MetaMarkers
  ct_scores = score_cells(log1p(cpm(sce_obj)), conserved_subclass_meta_markers %>% filter(rank <= num_markers))
  ct_enrichment = compute_marker_enrichment(ct_scores)
  #Max normalizing each cell-type enrichment
  ct_enrichment_norm = t(apply(ct_enrichment, 1, function(u) u/max(u)))
  
  #Computing the centroids
  ct_enrichment_norm  = as.data.frame(t(ct_enrichment_norm )) %>% mutate(clusters = sce_obj$cell_cluster)
  centroids = ct_enrichment_norm  %>% group_by(clusters) %>% summarize(across(which(colnames(ct_enrichment_norm)!= 'clusters'), mean)) %>% as.data.frame()
  #Get back into genes on rows and subclass on columns
  rownames(centroids) = centroids$clusters
  centroids = t(centroids[ ,2:ncol(centroids)])
  
  colnames(centroids) = paste(dataset_name, colnames(centroids), sep = '|')
  
  return(centroids)
}

```

```{r}

n_mark = 100

centroids_1 = get_cluster_MM_centroids(tasic18, 'tasic18', all_species_markers, num_markers = n_mark  )

centroids_2 = get_cluster_MM_centroids(zeng_10x_cells, 'zeng_10x_cells', all_species_markers, num_markers = n_mark )

centroids_3 = get_cluster_MM_centroids(zeng_10x_nuclei, 'zeng_10x_nuclei', all_species_markers, num_markers = n_mark )

centroids_4 = get_cluster_MM_centroids(zeng_smart_cells, 'zeng_smart_cells', all_species_markers, num_markers = n_mark )

centroids_5 = get_cluster_MM_centroids(zeng_smart_nuclei, 'zeng_smart_nuclei', all_species_markers, num_markers = n_mark  )

centroids_6 = get_cluster_MM_centroids(zeng_10xv2_wba, 'zeng_10xv2_wba', all_species_markers, num_markers = n_mark )

centroids_7 = get_cluster_MM_centroids(zeng_10xv3_wba, 'zeng_10xv3_wba', all_species_markers, num_markers = n_mark )

centroids_8 = get_cluster_MM_centroids(macosko_wba, 'macosko_wba', all_species_markers, num_markers = n_mark )




```

Various color schemes for the heatmap

```{r}
auroc_cols <- rev(grDevices::colorRampPalette(RColorBrewer::brewer.pal(11,"RdYlBu"))(100))

study_colors = MetBrewer::met.brewer("Veronese", n=8, type="continuous")
study_colors = study_colors[1:8] 
names(study_colors) = names(table(max_avg_mapping$study))

mn_auroc_col_fun = circlize::colorRamp2(c(.5, 1), c( "white", "red"))

```


```{r}


all_centroids = do.call('cbind', list(centroids_1, centroids_2, centroids_3, centroids_4, centroids_5, centroids_6, centroids_7, centroids_8))


#Make a row dendrogram across all the data, so it's the same ordering across the high and low orthology split
row_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-all_centroids), method = "ward.D2"))

#now make a high orthology heatmap and a low orthology heatmap

index = match(colnames(all_centroids), max_avg_mapping$mouse_mcluster)
clust_orth = max_avg_mapping$orthology[index]

orth_index = clust_orth == 'high_orthology'

high_col_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-t(all_centroids[ ,orth_index])), method = "ward.D2"))
low_col_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-t(all_centroids[ ,!orth_index])), method = "ward.D2"))

high_index = match(colnames(all_centroids[ , orth_index]), max_avg_mapping$mouse_mcluster)

clust_orth = max_avg_mapping$orthology[high_index]
study = max_avg_mapping$study[high_index]
primate_subclass = max_avg_mapping$primate_subclass[high_index]
mean_MN_subclass = max_avg_mapping$mean_subclass[high_index]
high_col_annot = HeatmapAnnotation(cluster_orthology = clust_orth, 
                                   mean_MN_subclass = mean_MN_subclass,
                                   study = study,
                                   primate_subclass = primate_subclass,
                                   show_annotation_name = F, 
                             col = list(cluster_orthology = c('high_orthology' = 'black', 'low_orthology' = 'grey'),
                                         study = study_colors,
                                        primate_subclass = celltype_color_palette,
                                        mean_MN_subclass = mn_auroc_col_fun))

high_heatmap = ComplexHeatmap::Heatmap(all_centroids[ ,orth_index], name = 'centroid', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(high_col_dendrogram), cluster_columns = high_col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 8),
                        top_annotation = high_col_annot,
                        show_column_names = F)


low_index = match(colnames(all_centroids[ , !orth_index]), max_avg_mapping$mouse_mcluster)

clust_orth = max_avg_mapping$orthology[low_index]
study = max_avg_mapping$study[low_index]
primate_subclass = max_avg_mapping$primate_subclass[low_index]
mean_MN_subclass = max_avg_mapping$mean_subclass[low_index]
low_col_annot = HeatmapAnnotation(cluster_orthology = clust_orth, 
                                  mean_MN_subclass = mean_MN_subclass,
                                   study = study,
                                   primate_subclass = primate_subclass,
                             col = list(cluster_orthology = c('high_orthology' = 'black', 'low_orthology' = 'grey'),
                                         study = study_colors,
                                        primate_subclass = celltype_color_palette,
                                        mean_MN_subclass = mn_auroc_col_fun))

low_heatmap = ComplexHeatmap::Heatmap(all_centroids[ ,!orth_index], name = 'centroid', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(low_col_dendrogram), cluster_columns = low_col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 8),
                        top_annotation = low_col_annot,
                        show_column_names = F)


comb_heatmap = draw(high_heatmap + low_heatmap)

pdf(file = '/home/werner/projects/mouse_brain_consensus_taxonomy/graphs/low_ortho/all_MM_centroid_scores.pdf', useDingbats = F, height = 6, width = 15)
comb_heatmap
dev.off()


```

The difference between the max cell-type markers and the next highest cell-type markers is a lot smaller in the low orthologous clusters
It's saying the same thing as the MetaNeighbor results really. The low orthologous clusters are more of a mix of different subclasses.


```{r}


cluster_vars = apply(all_centroids, 2, 'var' )
cluster_maxes = apply(all_centroids, 2, 'max' )
cluster_next_best = apply(all_centroids, 2, function(x) sort(x,partial=length(x)-1)[length(x)-1]  )

diff_max_and_next_best = cluster_maxes - cluster_next_best

index = match(colnames(all_centroids), max_avg_mapping$mouse_mcluster)

stat_df = data.frame(cluster = colnames(all_centroids), var_exp = cluster_vars, 
                     diff_next_best = diff_max_and_next_best,
                     orthology = max_avg_mapping$orthology[index], 
                     primate_subclass = max_avg_mapping$primate_subclass[index] )

ggplot(stat_df, aes(x = orthology, y = diff_next_best)) + geom_boxplot()
```


Load up the primate data, DO NOT exclude the non-consensus primate clusters
Looking to see if there are better matches for some of the mouse clusters in the non-consensus primate clusters


```{r}

human_10x = readRDS(file = '/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/primates/human_data_10x_gaba.rds')
human_ss = readRDS( file = '/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/primates/human_data_ss_gaba.rds')

chimp_10x = readRDS(file = '/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/primates/chimp_data_10x_gaba.rds')
chimp_ss = readRDS(file = '/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/primates/chimp_data_ss_gaba.rds')

gorilla_10x = readRDS(file = '/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/primates/gorilla_data_10x_gaba.rds')
gorilla_ss = readRDS(file = '/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/primates/gorilla_data_ss_gaba.rds')

macaca_10x = readRDS(file = '/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/primates/macaca_data_10x_gaba.rds')

marmoset_10x = readRDS(file = '/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/ortho_filt/primates/marmoset_data_10x_gaba.rds')



```


```{r}

primate_cluster_mappings = data.table::fread('../data/cross_species_mapping_cls86.csv')
primate_cluster_mappings

primate_cluster_mappings %>% filter(cross_species_cluster == 'lamp5_sncg_vip|3')

primate_cluster_mappings %>% filter(consensus_cluster_label == 'Pax6, Sncg|6')

```

Find the cross-species clusters where the confirmed_subclass in the data does not match up the subclass label in the cross-species-mapping table

human_10x and human_ss and gorilla_10x and gorilla_ss and macaca_10x:
lamp5_sncg_vip|7: confirmed_subclass is Pax6, subclass_label is Sncg
lamp5_sncg_vip|9: confirmed_subclass is Sncg, subclass_label is Vip

chimp_10x, chimp_ss:
same as human and gorilla and macaca
lamp5_sncg_vip|3: confirmed_subclass is Sncg, subclass_label is Lamp5

marmoset_10x:
same as human and gorilla and macaca
lamp5_sncg_vip|6: confirmed_subclass is Sncg, subclass_label is Pax6
```{r}

index = match(macaca_10x$cross_species_cluster, primate_cluster_mappings$cross_species_cluster)

macaca_10x$subclass_label = primate_cluster_mappings$subclass_label[index]
table(macaca_10x$subclass_label == macaca_10x$confirmed_subclass)
table(as.data.frame(colData(macaca_10x)) %>% filter(subclass_label != confirmed_subclass) %>% pull(cross_species_cluster))

index = macaca_10x$confirmed_subclass != macaca_10x$subclass_label
table(macaca_10x$confirmed_subclass[index], macaca_10x$subclass_label[index] )


```

Add the consensus cluster label
```{r}

index = match(human_10x$cross_species_cluster, primate_cluster_mappings$cross_species_cluster)
human_10x$cluster_label = primate_cluster_mappings$cluster_label[index]
human_10x$consensus_cluster_label = primate_cluster_mappings$consensus_cluster_label[index]
human_10x$consensus_cluster = primate_cluster_mappings$consensus_cluster[index]
na_index = is.na(human_10x$consensus_cluster_label)
human_10x$consensus_cluster_label[na_index] = human_10x$cluster_label[na_index]
human_10x$subclass_label = primate_cluster_mappings$subclass_label[index]
table(human_10x$subclass_label == human_10x$confirmed_subclass)
table(as.data.frame(colData(human_10x)) %>% filter(subclass_label != confirmed_subclass) %>% pull(cross_species_cluster))



index = match(human_ss$cross_species_cluster, primate_cluster_mappings$cross_species_cluster)
human_ss$cluster_label = primate_cluster_mappings$cluster_label[index]
human_ss$consensus_cluster_label = primate_cluster_mappings$consensus_cluster_label[index]
human_ss$consensus_cluster = primate_cluster_mappings$consensus_cluster[index]
na_index = is.na(human_ss$consensus_cluster_label)
human_ss$consensus_cluster_label[na_index] = human_ss$cluster_label[na_index]
human_ss$subclass_label = primate_cluster_mappings$subclass_label[index]
table(human_ss$subclass_label == human_ss$confirmed_subclass)
table(as.data.frame(colData(human_ss)) %>% filter(subclass_label != confirmed_subclass) %>% pull(cross_species_cluster))



index = match(chimp_10x$cross_species_cluster, primate_cluster_mappings$cross_species_cluster)
chimp_10x$cluster_label = primate_cluster_mappings$cluster_label[index]
chimp_10x$consensus_cluster_label = primate_cluster_mappings$consensus_cluster_label[index]
chimp_10x$consensus_cluster = primate_cluster_mappings$consensus_cluster[index]
na_index = is.na(chimp_10x$consensus_cluster_label)
chimp_10x$consensus_cluster_label[na_index] = chimp_10x$cluster_label[na_index]
chimp_10x$subclass_label = primate_cluster_mappings$subclass_label[index]
table(chimp_10x$subclass_label == chimp_10x$confirmed_subclass)
table(as.data.frame(colData(chimp_10x)) %>% filter(subclass_label != confirmed_subclass) %>% pull(cross_species_cluster))


index = match(chimp_ss$cross_species_cluster, primate_cluster_mappings$cross_species_cluster)
chimp_ss$cluster_label = primate_cluster_mappings$cluster_label[index]
chimp_ss$consensus_cluster_label = primate_cluster_mappings$consensus_cluster_label[index]
chimp_ss$consensus_cluster = primate_cluster_mappings$consensus_cluster[index]
na_index = is.na(chimp_ss$consensus_cluster_label)
chimp_ss$consensus_cluster_label[na_index] = chimp_ss$cluster_label[na_index]
chimp_ss$subclass_label = primate_cluster_mappings$subclass_label[index]
table(chimp_ss$subclass_label == chimp_ss$confirmed_subclass)
table(as.data.frame(colData(chimp_ss)) %>% filter(subclass_label != confirmed_subclass) %>% pull(cross_species_cluster))


index = match(gorilla_10x$cross_species_cluster, primate_cluster_mappings$cross_species_cluster)
gorilla_10x$cluster_label = primate_cluster_mappings$cluster_label[index]
gorilla_10x$consensus_cluster_label = primate_cluster_mappings$consensus_cluster_label[index]
gorilla_10x$consensus_cluster = primate_cluster_mappings$consensus_cluster[index]
na_index = is.na(gorilla_10x$consensus_cluster_label)
gorilla_10x$consensus_cluster_label[na_index] = gorilla_10x$cluster_label[na_index]
gorilla_10x$subclass_label = primate_cluster_mappings$subclass_label[index]
table(gorilla_10x$subclass_label == gorilla_10x$confirmed_subclass)
table(as.data.frame(colData(gorilla_10x)) %>% filter(subclass_label != confirmed_subclass) %>% pull(cross_species_cluster))


index = match(gorilla_ss$cross_species_cluster, primate_cluster_mappings$cross_species_cluster)
gorilla_ss$cluster_label = primate_cluster_mappings$cluster_label[index]
gorilla_ss$consensus_cluster_label = primate_cluster_mappings$consensus_cluster_label[index]
gorilla_ss$consensus_cluster = primate_cluster_mappings$consensus_cluster[index]
na_index = is.na(gorilla_ss$consensus_cluster_label)
gorilla_ss$consensus_cluster_label[na_index] = gorilla_ss$cluster_label[na_index]
gorilla_ss$subclass_label = primate_cluster_mappings$subclass_label[index]
table(gorilla_ss$subclass_label == gorilla_ss$confirmed_subclass)
table(as.data.frame(colData(gorilla_ss)) %>% filter(subclass_label != confirmed_subclass) %>% pull(cross_species_cluster))



index = match(macaca_10x$cross_species_cluster, primate_cluster_mappings$cross_species_cluster)
macaca_10x$cluster_label = primate_cluster_mappings$cluster_label[index]
macaca_10x$consensus_cluster_label = primate_cluster_mappings$consensus_cluster_label[index]
macaca_10x$consensus_cluster = primate_cluster_mappings$consensus_cluster[index]
na_index = is.na(macaca_10x$consensus_cluster_label)
macaca_10x$consensus_cluster_label[na_index] = macaca_10x$cluster_label[na_index]
macaca_10x$subclass_label = primate_cluster_mappings$subclass_label[index]
table(macaca_10x$subclass_label == macaca_10x$confirmed_subclass)
table(as.data.frame(colData(macaca_10x)) %>% filter(subclass_label != confirmed_subclass) %>% pull(cross_species_cluster))


index = match(marmoset_10x$cross_species_cluster, primate_cluster_mappings$cross_species_cluster)
marmoset_10x$cluster_label = primate_cluster_mappings$cluster_label[index]
marmoset_10x$consensus_cluster_label = primate_cluster_mappings$consensus_cluster_label[index]
marmoset_10x$consensus_cluster = primate_cluster_mappings$consensus_cluster[index]
na_index = is.na(marmoset_10x$consensus_cluster_label)
marmoset_10x$consensus_cluster_label[na_index] = marmoset_10x$cluster_label[na_index]
marmoset_10x$subclass_label = primate_cluster_mappings$subclass_label[index]
table(marmoset_10x$subclass_label == marmoset_10x$confirmed_subclass)
table(as.data.frame(colData(marmoset_10x)) %>% filter(subclass_label != confirmed_subclass) %>% pull(cross_species_cluster))

as.data.frame(colData(macaca_10x)) %>% filter(subclass_label != confirmed_subclass) %>% 
  group_by(confirmed_subclass, subclass_label) %>% summarise(n = n())


human_10x$subclass_label[human_10x$subclass_label == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
human_ss$subclass_label[human_ss$subclass_label == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
chimp_10x$subclass_label[chimp_10x$subclass_label == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
chimp_ss$subclass_label[chimp_ss$subclass_label == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
gorilla_10x$subclass_label[gorilla_10x$subclass_label == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
gorilla_ss$subclass_label[gorilla_ss$subclass_label == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
macaca_10x$subclass_label[macaca_10x$subclass_label == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
marmoset_10x$subclass_label[marmoset_10x$subclass_label == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'

human_10x$confirmed_subclass[human_10x$confirmed_subclass == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
human_ss$confirmed_subclass[human_ss$confirmed_subclass == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
chimp_10x$confirmed_subclass[chimp_10x$confirmed_subclass == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
chimp_ss$confirmed_subclass[chimp_ss$confirmed_subclass == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
gorilla_10x$confirmed_subclass[gorilla_10x$confirmed_subclass == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
gorilla_ss$confirmed_subclass[gorilla_ss$confirmed_subclass == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
macaca_10x$confirmed_subclass[macaca_10x$confirmed_subclass == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
marmoset_10x$confirmed_subclass[marmoset_10x$confirmed_subclass == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'


```


Check out the centroids of marker expression when looking over species

```{r}
all_primate_markers = list(
    human_10x = read_markers("metamarkers/markers_human_10x.csv.gz"),
    human_ss = read_markers("metamarkers/markers_human_ss.csv.gz"),
    chimp_10x = read_markers("metamarkers/markers_chimp_10x.csv.gz"),
    chimp_ss = read_markers("metamarkers/markers_chimp_ss.csv.gz"),
    gorilla_10x = read_markers("metamarkers/markers_gorilla_10x.csv.gz"),
    gorilla_ss = read_markers("metamarkers/markers_gorilla_ss.csv.gz"),
    macaca_10x = read_markers("metamarkers/markers_macaca_10x.csv.gz"),
    marmoset_10x= read_markers("metamarkers/markers_marmoset_10x.csv.gz")
)


all_primate_markers$human_10x$cell_type[all_primate_markers$human_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_primate_markers$human_ss$cell_type[all_primate_markers$human_ss$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_primate_markers$chimp_10x$cell_type[all_primate_markers$chimp_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_primate_markers$chimp_ss$cell_type[all_primate_markers$chimp_ss$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_primate_markers$gorilla_10x$cell_type[all_primate_markers$gorilla_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_primate_markers$gorilla_ss$cell_type[all_primate_markers$gorilla_ss$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_primate_markers$macaca_10x$cell_type[all_primate_markers$macaca_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_primate_markers$marmoset_10x$cell_type[all_primate_markers$marmoset_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'


all_conf_primate_markers = list(
    human_10x = read_markers("metamarkers/conf_markers_human_10x.csv.gz"),
    human_ss = read_markers("metamarkers/conf_markers_human_ss.csv.gz"),
    chimp_10x = read_markers("metamarkers/conf_markers_chimp_10x.csv.gz"),
    chimp_ss = read_markers("metamarkers/conf_markers_chimp_ss.csv.gz"),
    gorilla_10x = read_markers("metamarkers/conf_markers_gorilla_10x.csv.gz"),
    gorilla_ss = read_markers("metamarkers/conf_markers_gorilla_ss.csv.gz"),
    macaca_10x = read_markers("metamarkers/conf_markers_macaca_10x.csv.gz"),
    marmoset_10x= read_markers("metamarkers/conf_markers_marmoset_10x.csv.gz")
)


all_conf_primate_markers$human_10x$cell_type[all_conf_primate_markers$human_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_conf_primate_markers$human_ss$cell_type[all_conf_primate_markers$human_ss$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_conf_primate_markers$chimp_10x$cell_type[all_conf_primate_markers$chimp_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_conf_primate_markers$chimp_ss$cell_type[all_conf_primate_markers$chimp_ss$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_conf_primate_markers$gorilla_10x$cell_type[all_conf_primate_markers$gorilla_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_conf_primate_markers$gorilla_ss$cell_type[all_conf_primate_markers$gorilla_ss$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_conf_primate_markers$macaca_10x$cell_type[all_conf_primate_markers$macaca_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'
all_conf_primate_markers$marmoset_10x$cell_type[all_conf_primate_markers$marmoset_10x$cell_type == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'




```

This is using subclass labels from the supp. table to make subclass metamarkers
```{r}
auroc_cols <- rev(grDevices::colorRampPalette(RColorBrewer::brewer.pal(11,"RdYlBu"))(100))

#Compute the centoids
n_mark = 100

human_10x$cell_cluster = human_10x$consensus_cluster_label
centroids_human_10x = get_cluster_MM_centroids(human_10x, 'human_10x', 'human', all_primate_markers, num_markers = n_mark, species_filt = T  )
#Metadata for heatmap annotations
meta_human_10x = as.data.frame(colData(human_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

human_ss$cell_cluster = human_ss$consensus_cluster_label
centroids_human_ss = get_cluster_MM_centroids(human_ss, 'human_ss', 'human', all_primate_markers, num_markers = n_mark , species_filt = T  )
#Metadata for heatmap annotations
meta_human_ss = as.data.frame(colData(human_ss)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#chimp
chimp_10x$cell_cluster = chimp_10x$consensus_cluster_label
centroids_chimp_10x = get_cluster_MM_centroids(chimp_10x, 'chimp_10x', 'chimp', all_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_chimp_10x = as.data.frame(colData(chimp_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

chimp_ss$cell_cluster = chimp_ss$consensus_cluster_label
centroids_chimp_ss = get_cluster_MM_centroids(chimp_ss, 'chimp_ss', 'chimp', all_primate_markers, num_markers = n_mark , species_filt = T  )
#Metadata for heatmap annotations
meta_chimp_ss = as.data.frame(colData(chimp_ss)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#gorilla
gorilla_10x$cell_cluster = gorilla_10x$consensus_cluster_label
centroids_gorilla_10x = get_cluster_MM_centroids(gorilla_10x, 'gorilla_10x', 'gorilla', all_primate_markers, num_markers = n_mark , species_filt = T  )
#Metadata for heatmap annotations
meta_gorilla_10x = as.data.frame(colData(gorilla_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

gorilla_ss$cell_cluster = gorilla_ss$consensus_cluster_label
centroids_gorilla_ss = get_cluster_MM_centroids(gorilla_ss, 'gorilla_ss',  'gorilla', all_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_gorilla_ss = as.data.frame(colData(gorilla_ss)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#macaca
macaca_10x$cell_cluster = macaca_10x$consensus_cluster_label
centroids_macaca_10x = get_cluster_MM_centroids(macaca_10x, 'macaca_10x',  'macaca', all_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_macaca_10x = as.data.frame(colData(macaca_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(species_tech = 'macaca_10x') %>%
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#marmoset
marmoset_10x$cell_cluster = marmoset_10x$consensus_cluster_label
centroids_marmoset_10x = get_cluster_MM_centroids(marmoset_10x, 'marmoset_10x',  'marmoset', all_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_marmoset_10x = as.data.frame(colData(marmoset_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))



#combine centroids and get the row dendrogram across all columns
all_centroids = do.call('cbind', list(centroids_human_10x, centroids_human_ss, 
                                      centroids_chimp_10x, centroids_chimp_ss,
                                      centroids_gorilla_10x, centroids_gorilla_ss,
                                      centroids_macaca_10x, centroids_marmoset_10x))

row_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-all_centroids), method = "ward.D2"))

#Get the dataset metadata for the column annotations
index = match(colnames(centroids_human_10x), meta_human_10x$temp_label)
species_tech_human_10x = meta_human_10x$species_tech[index]
cluster_orth_human_10x = meta_human_10x$consensus_cluster[index]
subclass_human_10x = meta_human_10x$subclass_label[index]
conf_subclass_human_10x = meta_human_10x$confirmed_subclass[index]

index = match(colnames(centroids_human_ss), meta_human_ss$temp_label)
species_tech_human_ss = meta_human_ss$species_tech[index]
cluster_orth_human_ss = meta_human_ss$consensus_cluster[index]
subclass_human_ss = meta_human_ss$subclass_label[index]
conf_subclass_human_ss = meta_human_ss$confirmed_subclass[index]

index = match(colnames(centroids_chimp_10x), meta_chimp_10x$temp_label)
species_tech_chimp_10x = meta_chimp_10x$species_tech[index]
cluster_orth_chimp_10x = meta_chimp_10x$consensus_cluster[index]
subclass_chimp_10x = meta_chimp_10x$subclass_label[index]
conf_subclass_chimp_10x = meta_chimp_10x$confirmed_subclass[index]

index = match(colnames(centroids_chimp_ss), meta_chimp_ss$temp_label)
species_tech_chimp_ss = meta_chimp_ss$species_tech[index]
cluster_orth_chimp_ss = meta_chimp_ss$consensus_cluster[index]
subclass_chimp_ss = meta_chimp_ss$subclass_label[index]
conf_subclass_chimp_ss= meta_chimp_ss$confirmed_subclass[index]

index = match(colnames(centroids_gorilla_10x), meta_gorilla_10x$temp_label)
species_tech_gorilla_10x = meta_gorilla_10x$species_tech[index]
cluster_orth_gorilla_10x = meta_gorilla_10x$consensus_cluster[index]
subclass_gorilla_10x = meta_gorilla_10x$subclass_label[index]
conf_subclass_gorilla_10x = meta_gorilla_10x$confirmed_subclass[index]

index = match(colnames(centroids_gorilla_ss), meta_gorilla_ss$temp_label)
species_tech_gorilla_ss = meta_gorilla_ss$species_tech[index]
cluster_orth_gorilla_ss = meta_gorilla_ss$consensus_cluster[index]
subclass_gorilla_ss = meta_gorilla_ss$subclass_label[index]
conf_subclass_gorilla_ss = meta_gorilla_ss$confirmed_subclass[index]

index = match(colnames(centroids_macaca_10x), meta_macaca_10x$temp_label)
species_tech_macaca_10x = meta_macaca_10x$species_tech[index]
cluster_orth_macaca_10x = meta_macaca_10x$consensus_cluster[index]
subclass_macaca_10x = meta_macaca_10x$subclass_label[index]
conf_subclass_macaca_10x = meta_macaca_10x$confirmed_subclass[index]

index = match(colnames(centroids_marmoset_10x), meta_marmoset_10x$temp_label)
species_tech_marmoset_10x = meta_marmoset_10x$species_tech[index]
cluster_orth_marmoset_10x = meta_marmoset_10x$consensus_cluster[index]
subclass_marmoset_10x = meta_marmoset_10x$subclass_label[index]
conf_subclass_marmoset_10x = meta_marmoset_10x$confirmed_subclass[index]



all_species_tech = c(species_tech_human_10x, species_tech_human_ss, 
                     species_tech_chimp_10x, species_tech_chimp_ss,
                     species_tech_gorilla_10x, species_tech_gorilla_ss,
                     species_tech_macaca_10x, species_tech_marmoset_10x)
all_cluster_orth = c(cluster_orth_human_10x, cluster_orth_human_ss, 
                     cluster_orth_chimp_10x, cluster_orth_chimp_ss,
                     cluster_orth_gorilla_10x, cluster_orth_gorilla_ss,
                     cluster_orth_macaca_10x, cluster_orth_marmoset_10x)
all_subclass = c(subclass_human_10x, subclass_human_ss, 
                 subclass_chimp_10x, subclass_chimp_ss,
                 subclass_gorilla_10x, subclass_gorilla_ss,
                 subclass_macaca_10x, subclass_marmoset_10x)
all_conf_subclass = c(conf_subclass_human_10x, conf_subclass_human_ss, 
                 conf_subclass_chimp_10x, conf_subclass_chimp_ss,
                 conf_subclass_gorilla_10x, conf_subclass_gorilla_ss,
                 conf_subclass_macaca_10x, conf_subclass_marmoset_10x)

orth_index = all_cluster_orth == 'yes'

#Make a high orth and a low orth heatmap
high_col_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-t(all_centroids[ ,orth_index])), method = "ward.D2"))
low_col_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-t(all_centroids[ ,!orth_index])), method = "ward.D2"))



high_col_annot = HeatmapAnnotation(species_tech = all_species_tech[orth_index],
                              cluster_orth = all_cluster_orth[orth_index],
                              subclass = all_subclass[orth_index],
                              conf_subclass = all_conf_subclass[orth_index],
                              col = list(cluster_orth = c('yes' = 'black', 'no' = 'grey'),
                                         subclass = celltype_color_palette,
                                         conf_subclass = celltype_color_palette))


high_primate_heatmap = ComplexHeatmap::Heatmap(all_centroids[ ,orth_index], name = 'centroid', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(high_col_dendrogram), cluster_columns = high_col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 8),
                        show_column_names = T,
                        top_annotation = high_col_annot)

low_col_annot = HeatmapAnnotation(species_tech = all_species_tech[!orth_index],
                              cluster_orth = all_cluster_orth[!orth_index],
                              subclass = all_subclass[!orth_index],
                              conf_subclass = all_conf_subclass[!orth_index],
                              col = list(cluster_orth = c('yes' = 'black', 'no' = 'grey'),
                                         subclass = celltype_color_palette,
                                         conf_subclass = celltype_color_palette))


low_primate_heatmap = ComplexHeatmap::Heatmap(all_centroids[ ,!orth_index], name = 'centroid', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(low_col_dendrogram), cluster_columns = low_col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 8),
                        show_column_names = T,
                        top_annotation = low_col_annot)




primate_heatmap = draw(high_primate_heatmap + low_primate_heatmap)

pdf(file = '/home/werner/projects/mouse_brain_consensus_taxonomy/graphs/low_ortho/primate_MM_centroids.pdf', 
    useDingbats = F, height = 6, width = 15)
primate_heatmap 
dev.off()

```


This is using subclass labels from the data to make metamarkers, the confirmed_subclass in the metadata
```{r}
auroc_cols <- rev(grDevices::colorRampPalette(RColorBrewer::brewer.pal(11,"RdYlBu"))(100))

#Compute the centoids
n_mark = 100

human_10x$cell_cluster = human_10x$consensus_cluster_label
centroids_human_10x = get_cluster_MM_centroids(human_10x, 'human_10x',  'human', all_conf_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_human_10x = as.data.frame(colData(human_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

human_ss$cell_cluster = human_ss$consensus_cluster_label
centroids_human_ss = get_cluster_MM_centroids(human_ss, 'human_ss',  'human', all_conf_primate_markers, num_markers = n_mark , species_filt = T  )
#Metadata for heatmap annotations
meta_human_ss = as.data.frame(colData(human_ss)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#chimp
chimp_10x$cell_cluster = chimp_10x$consensus_cluster_label
centroids_chimp_10x = get_cluster_MM_centroids(chimp_10x, 'chimp_10x',  'chimp', all_conf_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_chimp_10x = as.data.frame(colData(chimp_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

chimp_ss$cell_cluster = chimp_ss$consensus_cluster_label
centroids_chimp_ss = get_cluster_MM_centroids(chimp_ss, 'chimp_ss',  'chimp', all_conf_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_chimp_ss = as.data.frame(colData(chimp_ss)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#gorilla
gorilla_10x$cell_cluster = gorilla_10x$consensus_cluster_label
centroids_gorilla_10x = get_cluster_MM_centroids(gorilla_10x, 'gorilla_10x',  'gorilla', all_conf_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_gorilla_10x = as.data.frame(colData(gorilla_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

gorilla_ss$cell_cluster = gorilla_ss$consensus_cluster_label
centroids_gorilla_ss = get_cluster_MM_centroids(gorilla_ss, 'gorilla_ss',  'gorilla',all_conf_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_gorilla_ss = as.data.frame(colData(gorilla_ss)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#macaca
macaca_10x$cell_cluster = macaca_10x$consensus_cluster_label
centroids_macaca_10x = get_cluster_MM_centroids(macaca_10x, 'macaca_10x',  'macaca',all_conf_primate_markers, num_markers = n_mark , species_filt = T  )
#Metadata for heatmap annotations
meta_macaca_10x = as.data.frame(colData(macaca_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(species_tech = 'macaca_10x') %>%
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#marmoset
marmoset_10x$cell_cluster = marmoset_10x$consensus_cluster_label
centroids_marmoset_10x = get_cluster_MM_centroids(marmoset_10x, 'marmoset_10x',  'marmoset', all_conf_primate_markers, num_markers = n_mark, species_filt = T   )
#Metadata for heatmap annotations
meta_marmoset_10x = as.data.frame(colData(marmoset_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))



#combine centroids and get the row dendrogram across all columns
all_centroids = do.call('cbind', list(centroids_human_10x, centroids_human_ss, 
                                      centroids_chimp_10x, centroids_chimp_ss,
                                      centroids_gorilla_10x, centroids_gorilla_ss,
                                      centroids_macaca_10x, centroids_marmoset_10x))

row_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-all_centroids), method = "ward.D2"))

#Get the dataset metadata for the column annotations
index = match(colnames(centroids_human_10x), meta_human_10x$temp_label)
species_tech_human_10x = meta_human_10x$species_tech[index]
cluster_orth_human_10x = meta_human_10x$consensus_cluster[index]
subclass_human_10x = meta_human_10x$subclass_label[index]
conf_subclass_human_10x = meta_human_10x$confirmed_subclass[index]

index = match(colnames(centroids_human_ss), meta_human_ss$temp_label)
species_tech_human_ss = meta_human_ss$species_tech[index]
cluster_orth_human_ss = meta_human_ss$consensus_cluster[index]
subclass_human_ss = meta_human_ss$subclass_label[index]
conf_subclass_human_ss = meta_human_ss$confirmed_subclass[index]

index = match(colnames(centroids_chimp_10x), meta_chimp_10x$temp_label)
species_tech_chimp_10x = meta_chimp_10x$species_tech[index]
cluster_orth_chimp_10x = meta_chimp_10x$consensus_cluster[index]
subclass_chimp_10x = meta_chimp_10x$subclass_label[index]
conf_subclass_chimp_10x = meta_chimp_10x$confirmed_subclass[index]

index = match(colnames(centroids_chimp_ss), meta_chimp_ss$temp_label)
species_tech_chimp_ss = meta_chimp_ss$species_tech[index]
cluster_orth_chimp_ss = meta_chimp_ss$consensus_cluster[index]
subclass_chimp_ss = meta_chimp_ss$subclass_label[index]
conf_subclass_chimp_ss= meta_chimp_ss$confirmed_subclass[index]

index = match(colnames(centroids_gorilla_10x), meta_gorilla_10x$temp_label)
species_tech_gorilla_10x = meta_gorilla_10x$species_tech[index]
cluster_orth_gorilla_10x = meta_gorilla_10x$consensus_cluster[index]
subclass_gorilla_10x = meta_gorilla_10x$subclass_label[index]
conf_subclass_gorilla_10x = meta_gorilla_10x$confirmed_subclass[index]

index = match(colnames(centroids_gorilla_ss), meta_gorilla_ss$temp_label)
species_tech_gorilla_ss = meta_gorilla_ss$species_tech[index]
cluster_orth_gorilla_ss = meta_gorilla_ss$consensus_cluster[index]
subclass_gorilla_ss = meta_gorilla_ss$subclass_label[index]
conf_subclass_gorilla_ss = meta_gorilla_ss$confirmed_subclass[index]

index = match(colnames(centroids_macaca_10x), meta_macaca_10x$temp_label)
species_tech_macaca_10x = meta_macaca_10x$species_tech[index]
cluster_orth_macaca_10x = meta_macaca_10x$consensus_cluster[index]
subclass_macaca_10x = meta_macaca_10x$subclass_label[index]
conf_subclass_macaca_10x = meta_macaca_10x$confirmed_subclass[index]

index = match(colnames(centroids_marmoset_10x), meta_marmoset_10x$temp_label)
species_tech_marmoset_10x = meta_marmoset_10x$species_tech[index]
cluster_orth_marmoset_10x = meta_marmoset_10x$consensus_cluster[index]
subclass_marmoset_10x = meta_marmoset_10x$subclass_label[index]
conf_subclass_marmoset_10x = meta_marmoset_10x$confirmed_subclass[index]



all_species_tech = c(species_tech_human_10x, species_tech_human_ss, 
                     species_tech_chimp_10x, species_tech_chimp_ss,
                     species_tech_gorilla_10x, species_tech_gorilla_ss,
                     species_tech_macaca_10x, species_tech_marmoset_10x)
all_cluster_orth = c(cluster_orth_human_10x, cluster_orth_human_ss, 
                     cluster_orth_chimp_10x, cluster_orth_chimp_ss,
                     cluster_orth_gorilla_10x, cluster_orth_gorilla_ss,
                     cluster_orth_macaca_10x, cluster_orth_marmoset_10x)
all_subclass = c(subclass_human_10x, subclass_human_ss, 
                 subclass_chimp_10x, subclass_chimp_ss,
                 subclass_gorilla_10x, subclass_gorilla_ss,
                 subclass_macaca_10x, subclass_marmoset_10x)
all_conf_subclass = c(conf_subclass_human_10x, conf_subclass_human_ss, 
                 conf_subclass_chimp_10x, conf_subclass_chimp_ss,
                 conf_subclass_gorilla_10x, conf_subclass_gorilla_ss,
                 conf_subclass_macaca_10x, conf_subclass_marmoset_10x)

orth_index = all_cluster_orth == 'yes'

#Make a high orth and a low orth heatmap
high_col_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-t(all_centroids[ ,orth_index])), method = "ward.D2"))
low_col_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-t(all_centroids[ ,!orth_index])), method = "ward.D2"))



high_col_annot = HeatmapAnnotation(species_tech = all_species_tech[orth_index],
                              cluster_orth = all_cluster_orth[orth_index],
                              subclass = all_subclass[orth_index],
                              conf_subclass = all_conf_subclass[orth_index],
                              col = list(cluster_orth = c('yes' = 'black', 'no' = 'grey'),
                                         subclass = celltype_color_palette,
                                         conf_subclass = celltype_color_palette))


high_primate_heatmap = ComplexHeatmap::Heatmap(all_centroids[ ,orth_index], name = 'centroid', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(high_col_dendrogram), cluster_columns = high_col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 8),
                        show_column_names = T,
                        top_annotation = high_col_annot)

low_col_annot = HeatmapAnnotation(species_tech = all_species_tech[!orth_index],
                              cluster_orth = all_cluster_orth[!orth_index],
                              subclass = all_subclass[!orth_index],
                              conf_subclass = all_conf_subclass[!orth_index],
                              col = list(cluster_orth = c('yes' = 'black', 'no' = 'grey'),
                                         subclass = celltype_color_palette,
                                         conf_subclass = celltype_color_palette))


low_primate_heatmap = ComplexHeatmap::Heatmap(all_centroids[ ,!orth_index], name = 'centroid', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(low_col_dendrogram), cluster_columns = low_col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 8),
                        show_column_names = T,
                        top_annotation = low_col_annot)




primate_heatmap = draw(high_primate_heatmap + low_primate_heatmap)

pdf(file = '/home/werner/projects/mouse_brain_consensus_taxonomy/graphs/low_ortho/conf_subclass_primate_MM_centroids.pdf', 
    useDingbats = F, height = 6, width = 15)
primate_heatmap 
dev.off()

```


This is using subclass labels from the supp. table to make subclass metamarkers
```{r}
auroc_cols <- rev(grDevices::colorRampPalette(RColorBrewer::brewer.pal(11,"RdYlBu"))(100))

#Compute the centoids
n_mark = 100

human_10x$cell_cluster = human_10x$consensus_cluster_label
centroids_human_10x = get_cluster_MM_centroids(human_10x, 'human_10x', 'human', all_species_markers, num_markers = n_mark  )
#Metadata for heatmap annotations
meta_human_10x = as.data.frame(colData(human_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

human_ss$cell_cluster = human_ss$consensus_cluster_label
centroids_human_ss = get_cluster_MM_centroids(human_ss, 'human_ss', 'human', all_species_markers, num_markers = n_mark  )
#Metadata for heatmap annotations
meta_human_ss = as.data.frame(colData(human_ss)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#chimp
chimp_10x$cell_cluster = chimp_10x$consensus_cluster_label
centroids_chimp_10x = get_cluster_MM_centroids(chimp_10x, 'chimp_10x', 'chimp', all_species_markers, num_markers = n_mark  )
#Metadata for heatmap annotations
meta_chimp_10x = as.data.frame(colData(chimp_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

chimp_ss$cell_cluster = chimp_ss$consensus_cluster_label
centroids_chimp_ss = get_cluster_MM_centroids(chimp_ss, 'chimp_ss', 'chimp', all_species_markers, num_markers = n_mark  )
#Metadata for heatmap annotations
meta_chimp_ss = as.data.frame(colData(chimp_ss)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#gorilla
gorilla_10x$cell_cluster = gorilla_10x$consensus_cluster_label
centroids_gorilla_10x = get_cluster_MM_centroids(gorilla_10x, 'gorilla_10x', 'gorilla', all_species_markers, num_markers = n_mark  )
#Metadata for heatmap annotations
meta_gorilla_10x = as.data.frame(colData(gorilla_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

gorilla_ss$cell_cluster = gorilla_ss$consensus_cluster_label
centroids_gorilla_ss = get_cluster_MM_centroids(gorilla_ss, 'gorilla_ss',  'gorilla', all_species_markers, num_markers = n_mark  )
#Metadata for heatmap annotations
meta_gorilla_ss = as.data.frame(colData(gorilla_ss)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#macaca
macaca_10x$cell_cluster = macaca_10x$consensus_cluster_label
centroids_macaca_10x = get_cluster_MM_centroids(macaca_10x, 'macaca_10x',  'macaca', all_species_markers, num_markers = n_mark  )
#Metadata for heatmap annotations
meta_macaca_10x = as.data.frame(colData(macaca_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(species_tech = 'macaca_10x') %>%
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))

#marmoset
marmoset_10x$cell_cluster = marmoset_10x$consensus_cluster_label
centroids_marmoset_10x = get_cluster_MM_centroids(marmoset_10x, 'marmoset_10x',  'marmoset', all_species_markers, num_markers = n_mark  )
#Metadata for heatmap annotations
meta_marmoset_10x = as.data.frame(colData(marmoset_10x)) %>% filter(!duplicated(consensus_cluster_label)) %>% 
  mutate(temp_label = paste(species_tech, consensus_cluster_label, sep = '|'))



#combine centroids and get the row dendrogram across all columns
all_centroids = do.call('cbind', list(centroids_human_10x, centroids_human_ss, 
                                      centroids_chimp_10x, centroids_chimp_ss,
                                      centroids_gorilla_10x, centroids_gorilla_ss,
                                      centroids_macaca_10x, centroids_marmoset_10x))

row_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-all_centroids), method = "ward.D2"))

#Get the dataset metadata for the column annotations
index = match(colnames(centroids_human_10x), meta_human_10x$temp_label)
species_tech_human_10x = meta_human_10x$species_tech[index]
cluster_orth_human_10x = meta_human_10x$consensus_cluster[index]
subclass_human_10x = meta_human_10x$subclass_label[index]
conf_subclass_human_10x = meta_human_10x$confirmed_subclass[index]

index = match(colnames(centroids_human_ss), meta_human_ss$temp_label)
species_tech_human_ss = meta_human_ss$species_tech[index]
cluster_orth_human_ss = meta_human_ss$consensus_cluster[index]
subclass_human_ss = meta_human_ss$subclass_label[index]
conf_subclass_human_ss = meta_human_ss$confirmed_subclass[index]

index = match(colnames(centroids_chimp_10x), meta_chimp_10x$temp_label)
species_tech_chimp_10x = meta_chimp_10x$species_tech[index]
cluster_orth_chimp_10x = meta_chimp_10x$consensus_cluster[index]
subclass_chimp_10x = meta_chimp_10x$subclass_label[index]
conf_subclass_chimp_10x = meta_chimp_10x$confirmed_subclass[index]

index = match(colnames(centroids_chimp_ss), meta_chimp_ss$temp_label)
species_tech_chimp_ss = meta_chimp_ss$species_tech[index]
cluster_orth_chimp_ss = meta_chimp_ss$consensus_cluster[index]
subclass_chimp_ss = meta_chimp_ss$subclass_label[index]
conf_subclass_chimp_ss= meta_chimp_ss$confirmed_subclass[index]

index = match(colnames(centroids_gorilla_10x), meta_gorilla_10x$temp_label)
species_tech_gorilla_10x = meta_gorilla_10x$species_tech[index]
cluster_orth_gorilla_10x = meta_gorilla_10x$consensus_cluster[index]
subclass_gorilla_10x = meta_gorilla_10x$subclass_label[index]
conf_subclass_gorilla_10x = meta_gorilla_10x$confirmed_subclass[index]

index = match(colnames(centroids_gorilla_ss), meta_gorilla_ss$temp_label)
species_tech_gorilla_ss = meta_gorilla_ss$species_tech[index]
cluster_orth_gorilla_ss = meta_gorilla_ss$consensus_cluster[index]
subclass_gorilla_ss = meta_gorilla_ss$subclass_label[index]
conf_subclass_gorilla_ss = meta_gorilla_ss$confirmed_subclass[index]

index = match(colnames(centroids_macaca_10x), meta_macaca_10x$temp_label)
species_tech_macaca_10x = meta_macaca_10x$species_tech[index]
cluster_orth_macaca_10x = meta_macaca_10x$consensus_cluster[index]
subclass_macaca_10x = meta_macaca_10x$subclass_label[index]
conf_subclass_macaca_10x = meta_macaca_10x$confirmed_subclass[index]

index = match(colnames(centroids_marmoset_10x), meta_marmoset_10x$temp_label)
species_tech_marmoset_10x = meta_marmoset_10x$species_tech[index]
cluster_orth_marmoset_10x = meta_marmoset_10x$consensus_cluster[index]
subclass_marmoset_10x = meta_marmoset_10x$subclass_label[index]
conf_subclass_marmoset_10x = meta_marmoset_10x$confirmed_subclass[index]



all_species_tech = c(species_tech_human_10x, species_tech_human_ss, 
                     species_tech_chimp_10x, species_tech_chimp_ss,
                     species_tech_gorilla_10x, species_tech_gorilla_ss,
                     species_tech_macaca_10x, species_tech_marmoset_10x)
all_cluster_orth = c(cluster_orth_human_10x, cluster_orth_human_ss, 
                     cluster_orth_chimp_10x, cluster_orth_chimp_ss,
                     cluster_orth_gorilla_10x, cluster_orth_gorilla_ss,
                     cluster_orth_macaca_10x, cluster_orth_marmoset_10x)
all_subclass = c(subclass_human_10x, subclass_human_ss, 
                 subclass_chimp_10x, subclass_chimp_ss,
                 subclass_gorilla_10x, subclass_gorilla_ss,
                 subclass_macaca_10x, subclass_marmoset_10x)
all_conf_subclass = c(conf_subclass_human_10x, conf_subclass_human_ss, 
                 conf_subclass_chimp_10x, conf_subclass_chimp_ss,
                 conf_subclass_gorilla_10x, conf_subclass_gorilla_ss,
                 conf_subclass_macaca_10x, conf_subclass_marmoset_10x)

orth_index = all_cluster_orth == 'yes'

#Make a high orth and a low orth heatmap
high_col_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-t(all_centroids[ ,orth_index])), method = "ward.D2"))
low_col_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-t(all_centroids[ ,!orth_index])), method = "ward.D2"))



high_col_annot = HeatmapAnnotation(species_tech = all_species_tech[orth_index],
                              cluster_orth = all_cluster_orth[orth_index],
                              subclass = all_subclass[orth_index],
                              conf_subclass = all_conf_subclass[orth_index],
                              col = list(cluster_orth = c('yes' = 'black', 'no' = 'grey'),
                                         subclass = celltype_color_palette,
                                         conf_subclass = celltype_color_palette))


high_primate_heatmap = ComplexHeatmap::Heatmap(all_centroids[ ,orth_index], name = 'centroid', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(high_col_dendrogram), cluster_columns = high_col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 8),
                        show_column_names = T,
                        top_annotation = high_col_annot)

low_col_annot = HeatmapAnnotation(species_tech = all_species_tech[!orth_index],
                              cluster_orth = all_cluster_orth[!orth_index],
                              subclass = all_subclass[!orth_index],
                              conf_subclass = all_conf_subclass[!orth_index],
                              col = list(cluster_orth = c('yes' = 'black', 'no' = 'grey'),
                                         subclass = celltype_color_palette,
                                         conf_subclass = celltype_color_palette))


low_primate_heatmap = ComplexHeatmap::Heatmap(all_centroids[ ,!orth_index], name = 'centroid', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(low_col_dendrogram), cluster_columns = low_col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 8),
                        show_column_names = T,
                        top_annotation = low_col_annot)




primate_heatmap = draw(high_primate_heatmap + low_primate_heatmap)

pdf(file = '/home/werner/projects/mouse_brain_consensus_taxonomy/graphs/low_ortho/all_mammal_MM_primate_centroids.pdf', 
    useDingbats = F, height = 6, width = 15)
primate_heatmap 
dev.off()

```








```{r}

as.data.frame(colData(chimp_10x))%>% 
  group_by(cross_species_cluster, consensus_cluster_label, subclass_label, cluster_label, cluster, confirmed_subclass) %>% summarise(n = n())


as.data.frame(colData(chimp_ss))%>% 
  group_by(cross_species_cluster, consensus_cluster_label, subclass_label, cluster_label, cluster, confirmed_subclass) %>% summarise(n = n())


```


```{r}

as.data.frame(colData(human_10x))%>% 
  group_by(cross_species_cluster, consensus_cluster_label, subclass_label, cluster_label, cluster, confirmed_subclass) %>% summarise(n = n())


as.data.frame(colData(human_ss))%>% 
  group_by(cross_species_cluster, consensus_cluster_label, subclass_label, cluster_label, cluster, confirmed_subclass) %>% summarise(n = n())


as.data.frame(colData(marmoset_10x))%>% 
  group_by(cross_species_cluster, consensus_cluster_label, subclass_label, cluster_label, cluster, confirmed_subclass) %>% summarise(n = n())


```


comparing to the CZI versions of the data

```{r}

human_czi = zellkonverter::readH5AD('/inkwell03/werner/mouse_brain_consensus_taxonomy/datasets/czi_primate_data/human_czi.h5ad')
human_czi


as.data.frame(colData(human_czi)) %>% filter(Neighborhood %in% c('lamp5_sncg_vip','sst_sst_chodl_pvalb')) %>% 
  group_by(assay) %>% summarise(n = n())

dim(human_10x)
dim(human_ss)

czi_human_10x_meta = as.data.frame(colData(human_czi)) %>% filter(Neighborhood %in% c('lamp5_sncg_vip','sst_sst_chodl_pvalb') &
                                                                    assay == "10x 3' v3")
czi_human_ss_meta = as.data.frame(colData(human_czi)) %>% filter(Neighborhood %in% c('lamp5_sncg_vip','sst_sst_chodl_pvalb') &
                                                                    assay == "Smart-seq v4")


table(rownames(czi_human_10x_meta) == human_10x$sample_id)
table(czi_human_10x_meta$nCount_RNA == human_10x$nCount_RNA)
table(czi_human_10x_meta$nFeature_RNA == human_10x$nFeature_RNA)



czi_human_10x_meta = czi_human_10x_meta %>% select(Cluster, Neighborhood, Subclass, CrossSpeciesCluster)
human_10x_meta = as.data.frame(colData(human_10x)) %>% select(neighborhood, confirmed_subclass, cluster, cluster_label, subclass_label, cross_species_cluster )


all_human_meta = cbind(czi_human_10x_meta, human_10x_meta  )


all_human_meta 

all_human_meta %>% group_by(CrossSpeciesCluster, cross_species_cluster) %>% summarise(n=n()) 

all_human_meta %>% group_by(Cluster, cluster) %>% summarise(n=n()) %>% 
  mutate(check_same = Cluster == cluster) %>% group_by(check_same) %>% summarise(n = n())

all_human_meta %>% group_by(Neighborhood, neighborhood) %>% summarise(n=n()) %>% 
  mutate(check_same = Neighborhood == neighborhood) %>% group_by(check_same) %>% summarise(n = n())

all_human_meta %>% group_by(Subclass, confirmed_subclass) %>% summarise(n=n()) %>% 
  mutate(check_same = Subclass == confirmed_subclass) %>% group_by(check_same) %>% summarise(n = n())

all_human_meta %>% group_by(Subclass, subclass_label) %>% summarise(n=n()) %>% 
  mutate(check_same = Subclass == subclass_label) %>% group_by(check_same) %>% summarise(n = n())


all_human_meta %>% group_by(cluster, confirmed_subclass) %>% summarise(n = n())

all_human_meta %>% group_by(cluster, cluster_label) %>% summarise(n = n())


```
















```{r}
#Human
#Change metdata columns to match other datasets
#human_10x$subclass_label = human_10x$confirmed_subclass
human_10x$cluster_label = human_10x$consensus_cluster_label

keep_meta = colData(human_10x)[ , c('subclass_label','cluster_label', 'consensus_cluster')]
colData(human_10x)= keep_meta

#Change metadata columns to match other datasets
#human_ss$subclass_label = human_ss$confirmed_subclass
human_ss$cluster_label = human_ss$consensus_cluster_label

keep_meta = colData(human_ss)[ , c('subclass_label','cluster_label', 'consensus_cluster')]
colData(human_ss)= keep_meta


#Chimp
#Change metdata columns to match other datasets
#chimp_10x$subclass_label = chimp_10x$confirmed_subclass
chimp_10x$cluster_label = chimp_10x$consensus_cluster_label

keep_meta = colData(chimp_10x)[ , c('subclass_label','cluster_label', 'consensus_cluster')]
colData(chimp_10x)= keep_meta

#Change metadata columns to match other datasets
#chimp_ss$subclass_label = chimp_ss$confirmed_subclass
chimp_ss$cluster_label = chimp_ss$consensus_cluster_label

keep_meta = colData(chimp_ss)[ , c('subclass_label','cluster_label', 'consensus_cluster')]
colData(chimp_ss)= keep_meta

#gorilla
#Change metdata columns to match other datasets
#gorilla_10x$subclass_label = gorilla_10x$confirmed_subclass
gorilla_10x$cluster_label = gorilla_10x$consensus_cluster_label

keep_meta = colData(gorilla_10x)[ , c('subclass_label','cluster_label', 'consensus_cluster')]
colData(gorilla_10x)= keep_meta

#Change metadata columns to match other datasets
#gorilla_ss$subclass_label = gorilla_ss$confirmed_subclass
gorilla_ss$cluster_label = gorilla_ss$consensus_cluster_label

keep_meta = colData(gorilla_ss)[ , c('subclass_label','cluster_label', 'consensus_cluster')]
colData(gorilla_ss)= keep_meta


#macaca
#Change metdata columns to match other datasets
#macaca_10x$subclass_label = macaca_10x$confirmed_subclass
macaca_10x$cluster_label = macaca_10x$consensus_cluster_label

keep_meta = colData(macaca_10x)[ , c('subclass_label','cluster_label', 'consensus_cluster')]
colData(macaca_10x)= keep_meta

#marmoset
#Change metdata columns to match other datasets
#marmoset_10x$subclass_label = marmoset_10x$confirmed_subclass
marmoset_10x$cluster_label = marmoset_10x$consensus_cluster_label

keep_meta = colData(marmoset_10x)[ , c('subclass_label','cluster_label', 'consensus_cluster')]
colData(marmoset_10x)= keep_meta


```



Now run MetaNeighbor with mouse clusters against primate clusters including the non-consensus clusters
```{r}

tasic18$subclass_label = rep(NA, length = ncol(tasic18))
tasic18$consensus_cluster = rep(NA, length = ncol(tasic18))
tasic18$cluster_label = tasic18$cell_cluster

zeng_10x_cells$subclass_label = rep(NA, length = ncol(zeng_10x_cells))
zeng_10x_cells$consensus_cluster = rep(NA, length = ncol(zeng_10x_cells))
zeng_10x_cells$cluster_label = zeng_10x_cells$cell_cluster

zeng_10x_nuclei$subclass_label = rep(NA, length = ncol(zeng_10x_nuclei))
zeng_10x_nuclei$consensus_cluster = rep(NA, length = ncol(zeng_10x_nuclei))
zeng_10x_nuclei$cluster_label = zeng_10x_nuclei$cell_cluster

zeng_smart_nuclei$subclass_label = rep(NA, length = ncol(zeng_smart_nuclei))
zeng_smart_nuclei$consensus_cluster = rep(NA, length = ncol(zeng_smart_nuclei))
zeng_smart_nuclei$cluster_label = zeng_smart_nuclei$cell_cluster

zeng_smart_cells$subclass_label = rep(NA, length = ncol(zeng_smart_cells))
zeng_smart_cells$consensus_cluster = rep(NA, length = ncol(zeng_smart_cells))
zeng_smart_cells$cluster_label = zeng_smart_cells$cell_cluster

zeng_10xv2_wba$subclass_label = rep(NA, length = ncol(zeng_10xv2_wba))
zeng_10xv2_wba$consensus_cluster = rep(NA, length = ncol(zeng_10xv2_wba))
zeng_10xv2_wba$cluster_label = zeng_10xv2_wba$cell_cluster

zeng_10xv3_wba$subclass_label = rep(NA, length = ncol(zeng_10xv3_wba))
zeng_10xv3_wba$consensus_cluster = rep(NA, length = ncol(zeng_10xv3_wba))
zeng_10xv3_wba$cluster_label = zeng_10xv3_wba$cell_cluster

macosko_wba$subclass_label = rep(NA, length = ncol(macosko_wba))
macosko_wba$consensus_cluster = rep(NA, length = ncol(macosko_wba))
macosko_wba$cluster_label = macosko_wba$cell_cluster
```





```{r}

all_mammal_sce = list(
  human_10x = human_10x,
  human_ss = human_ss,
  chimp_10x = chimp_10x, 
  chimp_ss = chimp_ss,
  gorilla_10x = gorilla_10x, 
  gorilla_ss = gorilla_ss,
  marmoset_10x = marmoset_10x,
  macaca_10x = macaca_10x,
  tasic18 = tasic18,
  zeng_10x_cells = zeng_10x_cells,
  zeng_10x_nuclei = zeng_10x_nuclei,
  zeng_smart_cells = zeng_smart_cells,
  zeng_smart_nuclei = zeng_smart_nuclei,
  zeng_10xv2_wba = zeng_10xv2_wba,
  zeng_10xv3_wba = zeng_10xv3_wba,
  macosko_wba = macosko_wba
)


```


Total primate and mouse cells is 595,300

```{r}

merged_mammal_sce = mergeSCE(all_mammal_sce)
dim(merged_mammal_sce)

rm(all_mammal_sce)
gc()

colData(merged_mammal_sce)[1:10, ]
```


Get highly variable genes
```{r}
start = Sys.time()
global_hvgs = variableGenes(dat = merged_mammal_sce, exp_labels = merged_mammal_sce$study_id, min_recurrence = 8)
length(global_hvgs)
Sys.time() - start

```

Use the top 1000
```{r}
global_hvgs = global_hvgs[1:1000]
```


```{r}
start = Sys.time()
all_mouse_clust_primate_clust_best_hits = MetaNeighborUS(var_genes = global_hvgs, 
                              dat = merged_mammal_sce, 
                              study_id = merged_mammal_sce$study_id,
                              cell_type = merged_mammal_sce$cluster_label,
                              fast_version = T,
                              one_vs_best = T, symmetric_output = F)


Sys.time() - start



pdf(file = '/home/werner/projects/mouse_brain_consensus_taxonomy/graphs/metaneighbor_plots/all_clusters_across_mammal_1vsbest.pdf',
    useDingbats = F, height = 25, width = 25)
plotHeatmap(all_mouse_clust_primate_clust_best_hits, cex = .2)
dev.off()

```



```{r}
celltype_color_palette = c("Chandelier"='#F641A8', 
                           "Pvalb"='#D93137', 
                           "Sst"='#FF9900', 
                           "Sst Chodl"='gold', 
                           "Lamp5"='#DA808C',
                           "Sncg"='#DF70FF',
                           "Vip"='#A45FBF', 
                           "Lamp5 Lhx6"='#935F50',
                           "Pax6"='#71238C',
                           'none' = 'grey')

```


prepping metadata for plotting

```{r}
all_metadata = as.data.frame(colData(merged_mammal_sce))


all_metadata$study_cluster = paste(all_metadata$study, gsub('|', '.', all_metadata$cluster_label, fixed = T), sep = '|')

primate_metadata = all_metadata %>% filter(grepl('human', study_id) | grepl('chimp', study_id) | grepl('gorilla', study_id) | grepl('macaca', study_id) | grepl('marmoset', study_id) ) %>% filter(!duplicated(study_cluster))


primate_metadata$subclass_label[primate_metadata$subclass_label == 'Lamp5_Lhx6'] = 'Lamp5 Lhx6'



```







```{r}

r_names = rownames(all_mouse_clust_primate_clust_best_hits)
r_index = grepl('human', r_names) | grepl('chimp', r_names) | grepl('gorilla', r_names) | grepl('macaca', r_names) | grepl('marmoset', r_names) 
c_index = !r_index

test_metasubclass_hits = all_mouse_clust_primate_clust_best_hits[r_index, c_index]

rowSums(is.na(test_metasubclass_hits)) == ncol(test_metasubclass_hits)


m = test_metasubclass_hits 
#Cluster first, need to swap NAs to 0 for clustering
m[is.na(m)] <- 0
row_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-m), method = "ward.D2"))
col_dendrogram <- stats::as.dendrogram(stats::hclust(dist(1-t(m)), method = "ward.D2"))


#And the primate subclass colors
index = match(rownames(test_metasubclass_hits), primate_metadata$study_cluster)

primate_subclass = primate_metadata$subclass_label[index]
primate_consensus = primate_metadata$consensus_cluster[index]

row_ha = rowAnnotation(primate_subclass = primate_subclass,
                       primate_consensus = primate_consensus,
                              col = list(primate_subclass = celltype_color_palette,
                                         primate_consensus = c('yes' = 'black', 'no' = 'grey')))


auroc_cols <- rev(grDevices::colorRampPalette(RColorBrewer::brewer.pal(11,"RdYlBu"))(100))

ht_mc = ComplexHeatmap::Heatmap(test_metasubclass_hits, name = 'AUROC', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(col_dendrogram), cluster_columns = col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 8), 
                        left_annotation = row_ha)
ht_mc = draw(ht_mc)


pdf(file = '/home/werner/projects/mouse_brain_consensus_taxonomy/graphs/metaneighbor_plots/all_clust_mouse_primate_heatmap.pdf',
    useDingbats = F, height = 20, width = 20)
ht_mc = ComplexHeatmap::Heatmap(test_metasubclass_hits, name = 'AUROC', col = auroc_cols,
                        row_order = order.dendrogram(row_dendrogram), cluster_rows = row_dendrogram,
                        column_order = order.dendrogram(col_dendrogram), cluster_columns = col_dendrogram,
                        column_names_gp = gpar(fontsize = 4),
                        row_names_gp = gpar(fontsize = 4), 
                        left_annotation = row_ha)
ht_mc = draw(ht_mc)


dev.off()




```












